src/sftp.rs

```diff
@@ -136,7 +136,7 @@ impl Sftp {
                 open_type as c_int,
             );
             if ret.is_null() {
-                Err(self.last_error())
+                Err(self.last_session_error().unwrap_or_else(|| { Error::unknown() }))
```

Comment:
Will the compiler allow you to use this instead:

```rust
Err(self.last_session_error().unwrap_or_else(Error::unknown))
```

Reply:
703b356